velox/exec/HashProbe.cpp

```diff
@@ -90,6 +90,8 @@ HashProbe::HashProbe(
       if (outIndex.value() != i) {
         isIdentityProjection = false;
       }
+    } else {
```

Comment:
Maybe we should write this code slightly differently to make the logic more clear :

 size_t countIdentityProjection = 0;
 for (auto i = 0; i < probeType->size(); ++i) {
    auto name = probeType->nameOf(i);
    auto outIndex = outputType_->getChildIdxIfExists(name);
    if (outIndex.has_value()) {
      identityProjections_.emplace_back(i, outIndex.value());
      if (outIndex.value() == i) {
        countIdentityProjection++;
      }
    }
  }
bool isIdentityProjection = countIdentityProjection == probeType.size() ? true : false;

Reply:
Thanks, @aditi-pandit ! Revised.